### PR TITLE
Adding possibility to select only date, only time or both with a mode property.

### PR DESCRIPTION
**Pull request by alex1712**
The biggest change would be the possibility to give to the picker a 'mode' property with one of these values ('time', 'date' or 'datetime'). With this property the picker will activate just the necessary elements.

I also added keys to the children in Month view and Year view as there were warnings because of that.

The last change was to use the class 'new' or 'old' in the day selector because there was a bug in which the user cannot select the days of the previous and the next month. Now I check if the target element of the click has one of those classes to set the previous or the next month on the date.

Thank you very much for all your work, I hope my changes help you to improve or at least to give you some new ideas.

Best regards

Alex

### CODE REVIEW

File: src/DateTimeField.jsx [hunk 1]
```diff
@@ -8,32 +8,46 @@ moment = require('moment');
...
-    defaultText: React.PropTypes.string
+    inputFormat: React.PropTypes.string,
+    defaultText: React.PropTypes.string,
+    mode: React.PropTypes.oneOf([Constants.MODE_DATE, Constants.MODE_DATETIME, Constants.MODE_TIME])
   },
   getDefaultProps: function() {
     return {
       dateTime: moment().format('x'),
       format: 'x',
-      inputFormat: "MM/DD/YY h:mm A",
```
Comment (chollier) [inline src/DateTimeField.jsx]: what motivated that ?

File: src/DateTimeField.jsx [hunk 2]
```diff
@@ -8,32 +8,46 @@ moment = require('moment');
...
       showToday: true,
       viewMode: 'days',
       daysOfWeekDisabled: [],
-      onChange: function (x) {
-        console.log(x);
-      }
```
Comment (chollier) [inline src/DateTimeField.jsx]: is there a specific reason to remove that ?

File: src/DateTimeField.jsx [hunk 3]
```diff
@@ -56,28 +70,33 @@ DateTimeField = React.createClass({
...
+      return this.props.onChange(moment(this.state.inputValue, this.state.inputFormat, true).format(this.props.format));
     });
 
   },
   setSelectedDate: function(e) {
+    var target = e.target;
+    var month;
+    if(target.className.includes("new")) month = this.state.viewDate.month() + 1;
+    else if(target.className.includes("old")) month = this.state.viewDate.month() - 1;
+    else month = this.state.viewDate.month();
```
Comment (chollier) [inline src/DateTimeField.jsx]: :+1:

**Comment by chollier** [comment]
I love it, i'm going to review it and maybe make a few changes (I want to update this.state.inputFormat on componentWillReceiveProps) and I'll merge it. Thanks a lot !

**Comment by alex1712** [comment]
Thank you very much for your comment. I am happy to hear that my changes are helpful.

Make, of course, all changes what you need :). I am looking forward to see the new version!

**Comment by chollier** [comment]
Merged :)

**Pull request merged**